Engine/ac/audiochannel.cpp

```diff
@@ -39,8 +39,11 @@ int AudioChannel_GetIsPlaying(ScriptAudioChannel *channel)
         return 0;
     }
 
-    if ((channels[channel->id] != NULL) &&
-        (channels[channel->id]->done == 0))
+    AudioChannelsLock _lock;
```

Comment:
wouldn't you need a global object (singleton) for this to work properly ?

Reply:
AudioChannelsLock internally uses a singleton mutex

Reply:
seems a bit verbose... maybe sometimes C-style non-OOP functions could be advantageous...

Reply:
no, RAII is advantageous

Reply:
Agreed, RAII for locks is extremely helpful.